### ISSUE TITLE
Add adblock filters

### ISSUE THREAD

**Issue by Kosay** [truefedex/tv-bro#37]
Hi
I tried many android TV browsers, and I think tv-bro and Samsung browser are the best (they are better than some paid browsers). I think it is better to allow the user to add more AdBlock lists or let the user can choose other filters. Another idea, can the user chose a mode (let us call it video tv mode) where it closes any other pup tabs.

Let me explain more if I open www.example.com to watch a video. When I click on the video, a new tab will open www.ad-example.com. 
I must close it to watch the video.
My idea in the Video TV will open only links in www.example.com and close any other out links.
If the user wants to go to a new website www.example2.com the user must turn off the video tv mode, then go to www.example2.com.

My best regards,

**Comment by MailYouLater** [comment]
I think, instead of having two separate modes the browser can operate in, it would be better to have it ask the user before opening any popups.

**Comment by syrumz** [comment]
> I think, instead of having two separate modes the browser can operate in, it would be better to have it ask the user before opening any popups.

Is this a thing? This is really really good when you dont want to open a pop up

**Comment by MailYouLater** [comment]
I've used a couple of browsers with popup blocking similar to what I'd like to see in TV Bro.
In one of the Android TV browsers (which I don't use because of other issues) it pops up a dialogue box to ask if you want to open the popup (with No preselected so that you're less likely to accidentally open an unwanted popup:
![Screenshot_Puffin TV Browser](https://user-images.githubusercontent.com/1279475/155423247-c9f6a590-b790-43eb-be7c-e3558106e62a.png)
In an app I sometimes use to cast content from my phone to a TV (usually only when other options aren't working for me) it briefly pops up a bar at the bottom of the screen to notify the user that a popup has been blocked, along with a button to open the popup instead.
![Screenshot_Cast to TV-XCast](https://user-images.githubusercontent.com/1279475/155423295-2812010a-15ed-4544-b3fe-d17a866a9c63.jpg)

I think it would be really nice if TV Bro used a bar that doesn't cover too much of the page's content when open, and if possible dismissed itself if the user doesn't interact with it after a set periood of time, even better (for me at least) if the bar is at the top so it's less likely to cover subtitles.  But even if it's just a big dialogue box that blocks half the screen until you manually choose if you want to block or allow the popup, it would still be better than opening the popup, after which the user has to open & navigate the menu to close it and go back to the tab they were on.

**Comment by Kosay** [comment]
Yes, it is a good idea. Or add an option (remember my choice) so if the user does not want to leave the website the browser will discard it directly.
This is easier special for TV users.

**Comment by MailYouLater** [comment]
> Or add an option (remember my choice) so if the user does not want to leave the website the browser will discard it directly.

In most cases I would agree that more options are better and options to remember a user's preference are good.  However, in this case, it would be easy for a user to assume that they always want to block popups, then later find that some functionality of a website is broken, likely not even realizing that they broke it by disallowing popups.  This is why what I recommend is a system which would  (preferably unobtrusively) block all popups, while also allowing the user an easy way to allow a desired popup through when the situation calls for it.

**Comment by Kosay** [comment]
> > Or add an option (remember my choice) so if the user does not want to leave the website the browser will discard it directly.
> 
> In most cases I would agree that more options are better and options to remember a user's preference are good. However, in this case, it would be easy for a user to assume that they always want to block popups, then later find that some functionality of a website is broken, likely not even realizing that they broke it by disallowing popups. This is why what I recommend is a system which would (preferably unobtrusively) block all popups, while also allowing the user an easy way to allow a desired popup through when the situation calls for it.

Yes, you are right if user choose "save my choice" to disallow popup but he forgets this option in another website. So, as I suggest, browser has two options
1) Disallow all popup
2) Ask user

**Comment by MailYouLater** [comment]
My point is that users will likely choose to disallow all popups, then think there's something wrong with the browser when some website is broken because it uses popups.  One thing we should never want is to introduce a 'feature' that breaks websites, drives away users, and causes people to submit invalid bug reports.

**Comment by Kosay** [comment]
The browser can show a sign to notify users about disallow popups. For example, write disallow mode on the browser title or add a tab for disallowing mode and it alarms the users about the mode.

**Comment by MailYouLater** [comment]
You don't seem to understand how the average human thinks.  There is no way to reliably inform them that the reason the button they just pushed didn't do what they expected because they changed a setting somewhere else, who-knows-how-many days, weeks or even months ago, except to pop something up _when it happens_.

If your going to pop up something to tell them when you block a popup, you may as well 
also offer an easy way to let the popup through, just this one time... but if you have it work that way, there's no need to add a setting to change this behavior unless you want one to turn popup blocking fully off.

Having a way to turn off the 'a popup has been blocked' info box while continuing to block all popups is just asking for trouble.

**Comment by truefedex** [comment]
Configurable adblock list was added to [v2.1.4](https://github.com/truefedex/tv-bro/releases/tag/v2.1.4)
